### PR TITLE
Enable SourceLink with embedded PDB

### LINKED ISSUE #4962: Enable SourceLink and embed PDB symbols for all NuGet packages
[SourceLink](https://github.com/dotnet/sourcelink) is essential and would make it easy to step _into_ Pulumi C# source supplied as NuGet packages.

Happy to submit a PR if someone can point me to how / where these packages are created at build-time.

### PR DESCRIPTION
**Pull request by gitfool**
Fixes #4962.

### CODE REVIEW

File: .github/workflows/windows-pr.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ jobs:
         go-version: [1.14.x]
         node-version: [10.x]
         python-version: [3.7]
-        dotnet: ['3.1.100']
+        dotnet: ['3.1.301']
```
Comment (mikhailshilkov) [inline .github/workflows/windows-pr.yml:13]: @stack72 @jaxxstorm any objections against this change?

**Comment by gitfool** [comment]
@mikhailshilkov I don't have a full development environment yet, pending the vs code dev container, so I didn't do a full build, just enough to verify the two `nupkg` artifacts had source link symbols.

image

I also bumped the .NET SDK to current latest since it helps with the build order when embedding untracked sources.

File: .github/workflows/windows-pr.yml (hunk 1 above)
Comment (stack72) [reply .github/workflows/windows-pr.yml:13]: Not from me - if you are happy with the new version being correct then we can roll that out everywhere

Comment (gitfool) [reply .github/workflows/windows-pr.yml:13]: See my comment above for why I bumped the .NET SDK.

Comment (mikhailshilkov) [reply .github/workflows/windows-pr.yml:13]: @stack72 Which versions are we using "everywhere" now? Could you link to the place where it's configured?

Comment (gitfool) [reply .github/workflows/windows-pr.yml:13]: https://github.com/dotnet/sourcelink/issues/572#issuecomment-583548551

Comment (stack72) [reply .github/workflows/windows-pr.yml:13]: In all of the providers (from a private repo) the version we have configured is 

```
dotnetversion: ['3.1.201'],
```

Comment (gitfool) [reply .github/workflows/windows-pr.yml:13]: @mikhailshilkov I just did a manual test and I can confirm I could step debug into Pulumi source from my program.

Comment (gitfool) [reply .github/workflows/windows-pr.yml:13]: image

**Pull request merged**